core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -122,6 +122,8 @@ trait Runtime[+R] { self =>
 
   protected abstract class UnsafeAPIV1 extends UnsafeAPI with UnsafeAPI3 {
 
+    private val fiberIdGen = self.fiberRefs.getOrDefault(FiberRef.currentFiberIdGenerator)
```

Comment:
shouldn't this be a def? the fiberRef can be overridden during fiber's execution

Reply:
We're within the `Runtime` trait here, and `self.fiberRefs` is immutable. The only way to "override" it here is by creating a new `Runtime`, and this val will be recreated in that case